presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformSpatialPredicates.java

```diff
@@ -331,9 +399,80 @@ else if (alignment < 0) {
                 newRightNode,
                 outputSymbols,
                 newFilter,
-                Optional.empty(),
-                Optional.empty(),
-                Optional.empty()));
+                leftPartitionSymbol,
+                rightPartitionSymbol,
+                kdbTree));
+    }
+
+    private static String loadKdbTree(String tableName, Session session, Metadata metadata, SplitManager splitManager, PageSourceManager pageSourceManager)
+    {
+        QualifiedObjectName name = toQualifiedObjectName(tableName, session.getCatalog().get(), session.getSchema().get());
+        TableHandle tableHandle = metadata.getTableHandle(session, name)
+                .orElseThrow(() -> new IllegalArgumentException(format("Table %s does not exist", name)));
```

Comment:
Categorize this better. If you don't want to create your own error category, specific for spatial join, you can at least throw a TableNotFoundException. Otherwise we will have increased internal error rate, when in fact this is user error.